### PR TITLE
[Refactor]: Refactor MAE

### PR DESCRIPTION
**Pull request by YuanLiuuuuuu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.
image

### CODE REVIEW

File: mmselfsup/models/algorithms/mae.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import torch
 
 from mmselfsup.data import SelfSupDataSample
-from mmselfsup.registry import MODELS
+from ..builder import MODELS
```
Comment (fangyixiao18) [inline mmselfsup/models/algorithms/mae.py]: better to use `from mmselfsup.registry import MODELS`

File: mmselfsup/models/backbones/mae_vit.py [hunk 2]
```diff
@@ -1,9 +1,10 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional, Sequence, Tuple, Union
+
 import torch
 from mmcls.models import VisionTransformer
-from torch import nn
 
-from mmselfsup.registry import MODELS
+from ..builder import MODELS
```
Comment (fangyixiao18) [inline mmselfsup/models/backbones/mae_vit.py]: The same as above

File: mmselfsup/models/backbones/mae_vit.py [hunk 3]
```diff
@@ -71,53 +74,45 @@ def __init__(self,
...
             init_cfg=init_cfg)
 
+        # position embedding is not learnable during pretraining
         self.pos_embed.requires_grad = False
         self.mask_ratio = mask_ratio
         self.num_patches = self.patch_resolution[0] * self.patch_resolution[1]
 
-    def init_weights(self):
+    def init_weights(self) -> None:
         super(MAEViT, self).init_weights()
```
Comment (fangyixiao18) [inline mmselfsup/models/backbones/mae_vit.py]:
```suggestion
        super().init_weights()
```

File: mmselfsup/models/backbones/mae_vit.py [hunk 4]
```diff
@@ -71,53 +74,45 @@ def __init__(self,
...
         Args:
-            x (torch.tensor): Image with data augmentation applied.
+            x (torch.Tensor): Image with data augmentation applied, which is
+                of shape B x L x C.
             mask_ratio (float): The mask ratio of total patches.
                 Defaults to 0.75.
 
         Returns:
-            tuple[Tensor, Tensor, Tensor]: masked image, mask and the ids
+            Tuple[torch.Tensor, Tensor, Tensor]: masked image, mask and the ids
```
Comment (fangyixiao18) [inline mmselfsup/models/backbones/mae_vit.py]:
```suggestion
            Tuple[torch.Tensor, torch.Tensor, torch.Tensor]: masked image, mask and the ids
```

File: mmselfsup/models/heads/mae_head.py [hunk 5]
```diff
@@ -31,7 +26,7 @@ def __init__(self,
         self.loss = MODELS.build(loss)
 
     def patchify(self, imgs: torch.Tensor) -> torch.Tensor:
-        """Get patches of the image."""
+
```
Comment (fangyixiao18) [inline mmselfsup/models/heads/mae_head.py:30]: Better to add some docstring here, indicate the shape of input tensor and output tensor

File: mmselfsup/models/backbones/mae_vit.py (hunk 3 above)
Comment (fangyixiao18) [reply mmselfsup/models/backbones/mae_vit.py]: please check other super init part

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: configs/selfsup/_base_/datasets/imagenet_mae.py [hunk 6]
```diff
@@ -2,14 +2,20 @@
 custom_imports = dict(imports='mmcls.datasets', allow_failed_imports=False)
 dataset_type = 'mmcls.ImageNet'
 data_root = 'data/imagenet/'
-file_client_args = dict(backend='disk')
+# file_client_args = dict(backend='disk')
+file_client_args = dict(
```
Comment (fangyixiao18) [inline configs/selfsup/_base_/datasets/imagenet_mae.py]: please change to `disk`

File: configs/selfsup/mae/mae_vit-base-p16_8xb512-coslr-1600e-fp16_in1k.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+_base_ = 'mae_vit-base-p16_8xb512-coslr-400e-fp16_in1k.py'
```
Comment (fangyixiao18) [inline configs/selfsup/mae/mae_vit-base-p16_8xb512-coslr-1600e-fp16_in1k.py]:
```suggestion
_base_ = 'mae_vit-base-p16_8xb512-fp16-coslr-400e_in1k.py'
```

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Review by fangyixiao18 — APPROVED** [review]

**Review by tonysy — APPROVED** [review]
LGTM

**Pull request merged**